tools/scripts/format-diff.ps1

```diff
@@ -0,0 +1,17 @@
+# Usage of this script needs git-clang-format.bat to be added to the
+# PATH.
+#
+# Usage:
+#
+#   format-diff.ps1
+#   format-diff.ps1 REF
+#
+# The former formats staged changes and the latter formats changes since
+# the REF commit.
+#
+# Note that unstaged changes will not be formatted.
+
+git-clang-format.bat `
+  --extensions h,c,cl --style "file:$PSScriptRoot\style.GNU" @args
+git-clang-format.bat `
+  --extensions hh,hpp,cc,cpp --style "file:$PSScriptRoot\style.CPP" @args
```

Comment:
Looking at the bash version of the script, the extensions should be ignored for anything under `lib/llvmopencl` and `lib/CL/devices/tce` and the CPP style should be used for everything there.

Reply:
Good catch, I'll work this out (may take time).